include/QBDI/VM.h

```diff
@@ -39,16 +40,16 @@ class InstrRule;
 struct MemCBInfo;
 
 class QBDI_EXPORT VM {
-    private:
+private:
     // Private internal engine
-    Engine*     engine;
+    std::unique_ptr<Engine> engine;
```

Comment:
Do we really need Engine as a pointer ? (i.e could it be a simple class attribute ?)

Reply:
VM object is a proxy to the Engine for the EndUser. If the VM have an instance of the Engine, we must provide Engine.h to the public API.